rs/utils/src/bloom_filter/bloom_filter.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use bitvec::prelude::*;
+
+#[derive(Hash)]
+pub struct HashKey {
+    user_id: u128,
+    doc_id: u128,
+}
+
+pub struct InMemoryBloomFilter<T: ?Sized> {
+    bits: BitVec,
+    num_hash_functions: usize,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T: ?Sized + Hash> InMemoryBloomFilter<T> {
+    pub fn new(expected_elements: usize, false_positive_prob: f64) -> Self {
```

Comment:
how do you know the size in mutable segment?

Reply:
In mutable segment it's gonna be the HashKey above right? So 2 u128 basically